server/src/main/java/org/opensearch/indices/replication/OngoingSegmentReplications.java

```diff
@@ -147,16 +147,15 @@ void startSegmentCopy(GetSegmentFilesRequest request, ActionListener<GetSegmentF
      */
     CopyState prepareForReplication(CheckpointInfoRequest request, FileChunkWriter fileChunkWriter) throws IOException {
         final CopyState copyState = getCachedCopyState(request.getCheckpoint());
-        if (allocationIdToHandlers.putIfAbsent(
+        // Clear (if any) existing handler
+        if (allocationIdToHandlers.containsKey(request.getTargetAllocationId())) {
+            logger.warn("Override handler for allocation id {}", request.getTargetAllocationId());
+            cancel(request.getTargetAllocationId(), "cancel due to retry");
+        }
+        allocationIdToHandlers.putIfAbsent(
```

Comment:
If we're cancelling any existing handlers, we could replace the putIfAbsent call by just a put.

Reply:
Thanks @Poojita-Raj for the comment, replaced with single compute call.